onnx/onnx-ml.proto

```diff
@@ -457,37 +441,31 @@ message ValueProto {
   }
 }
 
-// Define the types.
-message TypeProto {
-  // Defines a tensor shape. A dimension can be either an integer value
-  // or a symbolic variable. A symbolic variable represents an unknown
-  // dimension.
-  message TensorShapeProto {
-    message Dimension {
-      oneof value {
-        int64 dim_value = 1;
-        string dim_param = 2;   // namespace Shape
-      };
+// Defines a tensor shape. A dimension can be either an integer value
+// or a symbolic variable. A symbolic variable represents an unknown
+// dimension.
+message TensorShapeProto {
+  message Dimension {
+    oneof value {
+      int64 dim_value = 1;
+      string dim_param = 2;   // namespace Shape
     };
-    repeated Dimension dim = 1;
-  }
+  };
+  repeated Dimension dim = 1;
+}
 
-  message TensorTypeProto {
-    // This field MUST NOT have the value of UNDEFINED
-    // This field MUST be present for this version of the IR.
-    optional TensorProto.DataType elem_type = 1;
-    optional TensorShapeProto shape = 2;
-  }
+// Define the types.
+message TypeProto {
 
-  message SparseTensorTypeProto {
+  message Tensor {
     // This field MUST NOT have the value of UNDEFINED
     // This field MUST be present for this version of the IR.
     optional TensorProto.DataType elem_type = 1;
     optional TensorShapeProto shape = 2;
```

Comment:
Could we add a reserved properties block here, such as 100-200?

Reply:
In fact, systematically reserving blocks everywhere should help future-proof, I believe.